packages/site/cms-site/src/iframebridge/PreviewOverlay.tsx

```diff
@@ -22,14 +22,14 @@ export const PreviewOverlay = () => {
 
     return (
         <OverlayRoot style={{ height: bottomMostElementPosition }}>
-            {iFrameBridge.previewElementsData.map((element) => {
+            {iFrameBridge.previewElementsData.map((element, index) => {
                 const isSelected = element.adminRoute === iFrameBridge.selectedAdminRoute;
 
                 if (isSelected && lastSelectedElementPath !== element.adminRoute) {
                     lastSelectedElementPath = element.adminRoute;
                 }
 
-                return <PreviewOverlayElement key={element.adminRoute} element={element} />;
```

Comment:
Doesn't this cause other problems? For example, when moving elements around or adding a new element between existing elements?

I'm just thinking about this paragraph from the React docs and I think this might affect us here:

> You might be tempted to use an item’s index in the array as its key. In fact, that’s what React will use if you don’t specify a key at all. But the order in which you render items will change over time if an item is inserted, deleted, or if the array gets reordered. Index as a key often leads to subtle and confusing bugs.
https://react.dev/learn/rendering-lists#why-does-react-need-keys

Reply:
@nsams I'd also like your opinion on this

Reply:
I thought of using something like `element.adminRoute + "-" index` for exactly that reason but I tested it out with only `index` by removing/inserting/moving blocks and could not find any issues, every element was always rendered exactly where is was supposed to be rendered and there were no duplicate or missing elements.

Reply:
> @nsams I'd also like your opinion on this

I don't understand the underlaying problem, key is just a performance optimisation when children are reordered to reuse the correct already mounted component. Removing components (which is here the issue that it doesn't happen?) will work in any case.

Reply:
I don't understand react enough to explain why this happened, but the problem was that when multiple elements were rendered with the same key, on seemingly every re-render an additional element would be rendered and the previous element would not be removed from the dom. 

This is now solved with the unique key from the `index`. 

I believe the concern from @thomasdax98 is that e.g., the first preview-element (from the first block) is rendered with key `0`, when that block is removed another block's preview-element will have the key `0`. This could cause the preview-element to be rendered as the now deleted (previously first) block, instead of the block that is now the first. 

I couldn't find any issue like this during testing though, so I think this should be fine.